### ISSUE TITLE
No channel list displayed, because of missing 'default logo image'

### ISSUE THREAD

**Issue by levaidaniel** [ssapalski/plugin.audio.addict#8]
Error log with v0.5.0:
```
08:09:34.265 T:1309668256   DEBUG: -->Python Interpreter Initialized<--
08:09:34.265 T:1309668256   DEBUG: CPythonInvoker(104, /storage/.kodi/addons/plugin.audio.addict/addon.py): the source file to load is "/storage/.kodi/addons/plugin.audio.addict/addon.py"
08:09:34.266 T:1309668256   DEBUG: CPythonInvoker(104, /storage/.kodi/addons/plugin.audio.addict/addon.py): setting the Python path to /storage/.kodi/addons/plugin.audio.addict:/storage/.kodi/addons/script.module.requests/lib:/usr/lib/
python27.zip:/usr/lib/python2.7:/usr/lib/python2.7/plat-linux2:/usr/lib/python2.7/lib-tk:/usr/lib/python2.7/lib-old:/usr/lib/python2.7/lib-dynload:/usr/lib/python2.7/site-packages:/usr/lib/python2.7/site-packages/gtk-2.0
08:09:34.266 T:1309668256   DEBUG: CPythonInvoker(104, /storage/.kodi/addons/plugin.audio.addict/addon.py): entering source directory /storage/.kodi/addons/plugin.audio.addict
08:09:34.266 T:1309668256   DEBUG: CPythonInvoker(104, /storage/.kodi/addons/plugin.audio.addict/addon.py): instantiating addon using automatically obtained id of "plugin.audio.addict" dependent on version 2.20.0 of the xbmc.python api
08:09:34.525 T:1962037056   DEBUG: ------ Window Init (DialogBusy.xml) ------
08:09:37.378 T:1309668256   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.KeyError'>
                                            Error Contents: 'default'
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.audio.addict/addon.py", line 74, in <module>
                                                main()
                                              File "/storage/.kodi/addons/plugin.audio.addict/addon.py", line 70, in main
                                                audioaddict.run_addon(addon_url, addon_handle, addon_args)
                                              File "/storage/.kodi/addons/plugin.audio.addict/resources/lib/audioaddict/main.py", line 28, in run_addon
                                                main(addon, settings)
                                              File "/storage/.kodi/addons/plugin.audio.addict/resources/lib/audioaddict/main.py", line 47, in main
                                                show_channels(addon, settings)
                                              File "/storage/.kodi/addons/plugin.audio.addict/resources/lib/audioaddict/channels.py", line 13, in show_channels
                                                add_channels_to_kodi_directory(addon, settings)
                                              File "/storage/.kodi/addons/plugin.audio.addict/resources/lib/audioaddict/channels.py", line 40, in add_channels_to_kodi_directory
                                                list_item = create_list_item(channel)
                                              File "/storage/.kodi/addons/plugin.audio.addict/resources/lib/audioaddict/channels.py", line 59, in create_list_item
                                                image_url = channel.image_default()
                                              File "/storage/.kodi/addons/plugin.audio.addict/resources/lib/audioaddict/api.py", line 74, in image_default
                                                url = "http:%s" % self._channel['images']['default']
                                            KeyError: 'default'
                                            -->End of Python script error report<--
08:09:37.397 T:1309668256    INFO: Python script stopped
08:09:37.397 T:1309668256   DEBUG: Thread LanguageInvoker 1309668256 terminating
08:09:37.415 T:1962037056   ERROR: GetDirectory - Error getting plugin://plugin.audio.addict/?listen_key=[...]&mode=open_network&network_key=di
08:09:37.415 T:1962037056   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.audio.addict/?listen_key=[...]&mode=open_network&network_key=di) failed
08:09:37.426 T:1962037056   DEBUG: [Warning] CGUITextureManager::GetTexturePath: could not find texture 'DefaultAddonMusic.png'
08:09:37.431 T:1962037056   DEBUG: [Warning] CGUITextureManager::GetTexturePath: could not find texture 'DefaultMusicPlaylists.png'
08:09:37.433 T:1962037056   DEBUG: [Warning] CGUITextureManager::GetTexturePath: could not find texture 'DefaultFolder.png'
08:09:37.434 T:1962037056   DEBUG: CGUIMediaWindow::GetDirectory ()
08:09:37.434 T:1962037056   DEBUG:   ParentPath = [plugin://plugin.audio.addict/?listen_key=[...]&mode=open_network&network_key=di]
08:09:37.443 T:1962037056   DEBUG: [Warning] CGUITextureManager::GetTexturePath: could not find texture 'DefaultAddonMusic.png'
08:09:37.447 T:1962037056   DEBUG: [Warning] CGUITextureManager::GetTexturePath: could not find texture 'DefaultMusicPlaylists.png'
08:09:37.449 T:1962037056   DEBUG: [Warning] CGUITextureManager::GetTexturePath: could not find texture 'DefaultFolder.png'
08:09:37.452 T:1962037056   DEBUG: RetrieveMusicInfo() took 0 msec
08:09:37.463 T:1309668256   DEBUG: Thread BackgroundLoader start, auto delete: false
08:09:37.471 T:1309668256   DEBUG: Thread BackgroundLoader 1309668256 terminating
```

**Comment by levaidaniel** [comment]
So basically you need to have more exception handling in your code because you're expecting eg. that every channel has a 'logo' image associated to it. Now that just so happens, the 'Synthwave' channel does not.
This simple diff will put the DI.FM logo in place of a missing default channel logo:
```diff
--- resources/lib/audioaddict/api.py.orig       2017-11-05 10:58:16.000000000 +0100
+++ resources/lib/audioaddict/api.py    2017-11-05 11:37:43.000000000 +0100
@@ -71,7 +71,10 @@ class Channel(object):
         self._channel = parsed_json

     def image_default(self):
-        url = "http:%s" % self._channel['images']['default']
+        try:
+                url = "http:%s" % self._channel['images']['default']
+        except KeyError:
+                url = "http://cdn.audioaddict.com/di.fm/assets/logo-opengraph-5615da4c38b1a8f3a275a741f35a0894.png"
         url = url.split('{')[0]

         return url
```

Boom. Add-on working again.
Pretty rudimentary, but _you get the picture_. (see what I did there? :-) )

**Comment by ssapalski** [comment]
The Synthwave channel doesn't appear anymore on the web site and Android App. That looks more like it got disabled on purpose and I have to update the logic to filter disabled channels instead of showing a disabled channel with a default image.

I will look more closely into this issue!

**Comment by levaidaniel** [comment]
Fair enough, thanks!

**Comment by ssapalski** [comment]
I can confirm that the Synthwave channel is disabled, the playlist of this channel is empty.
This is interesting since other disabled channels are marked differently in the JSON response of the channel list.

Maybe this channel is only temporarily disabled. Anyway, fix will be released today!

**Comment by ssapalski** [comment]
Fixed with: 1713e4838743203d23ad7f26986f3c31890435ca
